### DISCUSSION TITLE
Inav 7.1.0 release code- compilation issues- build using vscode not successful. Build using MS Visual Studio 2022 worked but further build failed. Seems like compliation issue.

### DISCUSSION THREAD

**Discussion by PredySoft** [iNavFlight/inav#9786] [Q&A] [answered]
Hello team I would like to contribute with small functionality for myself, but if it is not causing any issues may be wider audience could benefit from that. But I struggle to compile the application and test it loclally. I am facing Inav compilation issues- build using vscode is not successful. Build using MS Visual Studio 2022 worked but further build using ninja failed. Which compiler are you using to make reliable build of the application and run your beta version on local? Do you have any tip on how to fix that. No change in the filestructure from git repo was done from my side. I downloaded git repo in the zip package and extracted locally in the inav7 project folder. Thanks a lot for your help.

vscode error example:
[cmake] CMake Error at cmake/stm32.cmake:242 (add_executable):
[cmake]   CMake doesn't support IPO for current compiler
[cmake] Call Stack (most recent call first):
[cmake]   cmake/stm32.cmake:374 (add_stm32_executable)
[cmake]   cmake/stm32f7.cmake:91 (target_stm32)
[cmake]   cmake/stm32f7.cmake:126 (target_stm32f7xx)
[cmake]   src/main/target/ZEEZF7/CMakeLists.txt:3 (target_stm32f722xe)
[cmake] 
[cmake] 
[cmake] -- Generating done (15.3s)
[cmake] CMake Generate step failed.  Build files cannot be regenerated correctly.
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -SC:/repos/inav7 -Bc:/repos/inav7/build -G "Visual Studio 17 2022" -T host=x64 -A x64 exited with code: 1

visual studio fail example:
PS C:\repos\inav7\out\build\x64-Debug> ninja
[1/156198] Generating OMNIBUSF7NANOV7/settings_generated.h, OMNIBUSF7NANOV7/settings_generated.c
FAILED: src/main/target/AIRBOTF7/OMNIBUSF7NANOV7/settings_generated.h src/main/target/AIRBOTF7/OMNIBUSF7NANOV7/settings_generated.c C:/repos/inav7/out/build/x64-Debug/src/main/target/AIRBOTF7/OMNIBUSF7NANOV7/settings_generated.h C:/repos/inav7/out/build/x64-Debug/src/main/target/AIRBOTF7/OMNIBUSF7NANOV7/settings_generated.c
C:\Windows\system32\cmd.exe /C "cd /D C:\repos\inav7\out\build\x64-Debug\src\main\target\AIRBOTF7 && "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E env CFLAGS="-ffunction-sections -fdata-sections -fno-common -mthumb -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-sp-d16 -fsingle-precision-constant -Wdouble-promotion -Wall -Wextra -Wunsafe-loop-optimizations -Wdouble-promotion -Wstrict-prototypes -Werror=switch -IC:/repos/inav7/src/main/target/AIRBOTF7 -IC:/repos/inav7/lib/main/STM32F7/Drivers/STM32F7xx_HAL_Driver/Inc -IC:/repos/inav7/lib/main/STM32F7/Drivers/CMSIS/Device/ST/STM32F7xx/Include -IC:/repos/inav7/lib/main/STM32F7/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -IC:/repos/inav7/lib/main/STM32F7/Middlewares/ST/STM32_USB_Device_Library/Class/CDC/Inc -IC:/repos/inav7/lib/main/STM32F7/Middlewares/ST/STM32_USB_Device_Library/Class/HID/Inc -IC:/repos/inav7/lib/main/STM32F7/Middlewares/ST/STM32_USB_Device_Library/Class/CDC_HID/Inc -IC:/repos/inav7/lib/main/STM32F7/Middlewares/ST/STM32_USB_Device_Library/Class/MSC/Inc -IC:/repos/inav7/src/main/vcp_hal -IC:/repos/inav7/lib/main/CMSIS/Core/Include -IC:/repos/inav7/lib/main/CMSIS/DSP/Include -IC:/repos/inav7/src/main/target -IC:/repos/inav7/out/build/x64-Debug/src/main/target/AIRBOTF7/OMNIBUSF7NANOV7 -IC:/repos/inav7/lib -IC:/repos/inav7/src/main -IC:/repos/inav7/lib/main/MAVLink -DFC_VERSION_MAJOR=7 -DFC_VERSION_MINOR=1 -DFC_VERSION_PATCH_LEVEL=0 -DUSE_USB_MSC -DHSE_VALUE=8000000 -D__FPU_PRESENT=1 -DARM_MATH_CM7 -DARM_MATH_MATRIX_CHECK -DARM_MATH_ROUNDING -DUNALIGNED_SUPPORT_DISABLE -DUSE_HAL_DRIVER -DUSE_FULL_LL_DRIVER -DSTM32F7 -DSTM32F722xx -DSTM32F722xE -DMCU_FLASH_SIZE=512 -D__FORKNAME__=inav -D__REVISION__="GITDIR-N" -D__TARGET__="OMNIBUSF7NANOV7" -DOMNIBUSF7NANOV7" TARGET=OMNIBUSF7NANOV7 PATH="C:/repos/inav7/tools/gcc-arm-none-eabi-10.3-2021.10/bin:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\bin\HostX64\x64 "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\VCPackages" "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow" "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer" "C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\bin\Roslyn" "C:\Program Files\Microsoft Visual Studio\2022\Community\Team Tools\DiagnosticsHub\Collector" "C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\\x64" "C:\Program Files (x86)\Windows Kits\10\bin\\x64" "C:\Program Files\Microsoft Visual Studio\2022\Community\\MSBuild\Current\Bin\amd64" C:\Windows\Microsoft.NET\Framework64\v4.0.30319 "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools;C:\Ruby32-x64\bin" C:\Windows\system32 C:\Windows C:\Windows\System32\Wbem "C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files\dotnet;C:\Program Files\CMake\bin" "C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\13.2 Rel1\bin" C:\Users\admin\AppData\Local\Microsoft\WindowsApps "C:\Users\admin\AppData\Local\Programs\Microsoft VS Code\bin" C:\Users\admin\.dotnet\tools C:\Ruby32-x64\bin "C:\Program Files\CMake\bin" "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin" "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja" "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe" C:\Program Files\Microsoft Visual Studio\2022\Community\VC\vcpkg" SETTINGS_CXX= C:/Ruby32-x64/bin/ruby.exe C:/repos/inav7/src/utils/settings.rb C:/repos/inav7 C:/repos/inav7/src/main/fc/settings.yaml -o C:/repos/inav7/out/build/x64-Debug/src/main/target/AIRBOTF7/OMNIBUSF7NANOV7"
no such file or directory

**Comment by DzikuVx** [comment, marked as answer]
Building directly in windows is not supported. Please use WSL approach https://github.com/iNavFlight/inav/blob/master/docs/development/Building%20in%20Windows%2010%20or%2011%20with%20Linux%20Subsystem.md

**Reply by PredySoft** [reply]
Hi I get the point and discovered this document in the meantime sooner today. That is working well.
